web/src/Storage.jsx

```diff
@@ -31,47 +33,57 @@ export default function Storage() {
   const [state, dispatch] = useReducer(reducer, {
     targets: [],
     target: "",
-    proposal: []
+    actions: [],
+    error: false
   });
-  const { target, targets, proposal } = state;
+  const { target, targets, actions, error } = state;
 
   const onAccept = selected =>
     client
-      .setOption("Disk", selected)
-      .then(() => dispatch({ type: "CHANGE_TARGET", payload: selected }));
+      .calculateStorageProposal({ candidateDevices: [selected] })
+      .then(result => {
+        const payload = { selected, error: (result !== 0) };
+        dispatch({ type: "CHANGE_TARGET", payload });
+      });
 
   useEffect(async () => {
-    const proposal = await client.getStorage();
-    const disk = await client.getOption("Disk");
-    const disks = await client.getDisks();
+    const { availableDevices: disks, candidateDevices: [disk] } = await client.getStorageProposal();
+    const actions = await client.getStorageActions();
     dispatch({
       type: "LOAD",
-      payload: { target: disk, targets: disks, proposal }
+      payload: { target: disk, targets: disks, actions }
     });
   }, []);
 
   useEffect(() => {
     // TODO: abstract D-Bus details
     return client.onPropertyChanged((_path, _iface, _signal, args) => {
-      const [_, changes] = args;
-      if (Object.keys(changes).includes("Disk")) {
-        client
-          .getStorage()
-          .then(proposal =>
-            dispatch({ type: "UPDATE_PROPOSAL", payload: proposal })
-          );
+      const [iface, properties] = args;
+
+      if (iface !== "org.opensuse.DInstaller.Storage.Actions1") {
+        return;
       }
+
+      const newActions = properties.All.v.map(action => {
+        const { Text: textVar, Subvol: subvolVar } = action.v;
+        return { text: textVar.v, subvol: subvolVar.v };
+      });
```

Comment:
I'd move that mapping to an util function.

Reply:
I prefer to wait until we improve `InstallerClient`. In the component, you shouldn't need to deal with D-Bus variants.